docs/examples/patterns/recursion.md

```diff
@@ -0,0 +1,106 @@
+# Recursion
+
+Recursion is the workhorse of Prometheux, and is a pattern that occurs naturally
+in many situations. In all recursive cases, there is :
+
+1. a recursive case, where a predicate is derived by itself, and
+2. a base case, which must not be derivable by itself, allowing the recursion to terminate.
+
+Consider a graph of nodes and edges between them. A path can be defined as:
+
+1. A simple edge between 2 nodes.
+2. The list of such edges between any 2 nodes, via N-2 other nodes.
+
+```prolog showLineNumbers
+path(Start, End) :- edge(Start, End). % Base case
+path(Start, End) :- path(Start, Via), edge(Via, End). % Recursive case
+```
+
+See how `path` can be a simple hop from one node to another, or requires
+2 hops (via 1 intermediate node), or via 3 hops (via 2 intermediate nodes).
+
+### Transitive Closure
+
+To find all the pairwise connections between all others nodes in a graph:
+
+// add aimage here
```

Comment:
we could use the one in SLIDE 17 in [this](https://docs.google.com/presentation/d/1rXLHhKxll0tTIaFTDg67_74edLXQEkH2ei1nvu6xOVk/edit?usp=sharing) presentation